md2conf/application.py

```diff
@@ -15,16 +23,70 @@ def update_document(
     api.update_page(document.page_id, document.xhtml())
 
 
+
+def synchronize_pages(api: ConfluenceSession, path: str, options: ConfluenceDocumentOptions) -> None:
+    if os.path.isdir(path):
+        synchronize_directory(api, path, options)
+    elif os.path.isfile(path):
+        synchronize_page(api, path, options)
+    else:
+        raise ValueError(f"expected: valid file of directory path. got: {path}")
+
+
 def synchronize_page(
-    api: ConfluenceSession, path: str, options: ConfluenceDocumentOptions
+    api: ConfluenceSession, path: str, options: ConfluenceDocumentOptions, page_metadata: dict[str, str] = dict()
 ) -> None:
     page_path = os.path.abspath(path)
     base_path = os.path.dirname(page_path)
 
-    document = ConfluenceDocument(path, options)
+    LOGGER.info(f"synchronize_page: {page_path}")
+    document = ConfluenceDocument(path, options, page_metadata)
```

Comment:
Maybe it's an oversight on my end but does this line up with the type annotation above and the declared signature of `ConfluenceDocument`? Can you run [mypy](https://mypy.readthedocs.io/en/stable/) on the source to check?